### PR TITLE
Fix crashes in skymatch when some footprints almost entirely overlap

### PR DESCRIPTION
**Pull request by mcara**
This PR essentially re-implements enhancement in `stsci.skypac` in https://github.com/spacetelescope/stsci.skypac/pull/49

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
 ---
 
-- Fixed indentation bug in ami_analyze, so now all results are sufficiently 
-  close to the results of the stand-alone prototype. Other modifications include 
-  minor tweaks to more closely match those in the prototype code: changed some of 
-  initial values of the estimation parameters, and the filtering routine 
+- Fixed indentation bug in ami_analyze, so now all results are sufficiently
+  close to the results of the stand-alone prototype. Other modifications include
+  minor tweaks to more closely match those in the prototype code: changed some of
+  initial values of the estimation parameters, and the filtering routine
```
Comment (mcara) [inline CHANGES.rst]: @jdavies-st  I have no idea where this came from...

Comment (mcara) [reply CHANGES.rst]: @jdavies-st Ah, I know. My editor strips ending white spaces.

**Comment by mcara** [comment]
yes. This avoids attempting to compute the intersection and then run into the roundoff issues.

**Comment by nden** [comment]
Seems uncontroversial, merging.

**Pull request merged**